src/components/MultiShopSelection.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from "react"
+import {AllItems} from "./ItemSelection"
+import { Items, itemRarities } from "../utilities/types"
+
+const MultiShopSelection: React.FC<{allItems: AllItems, handleItemSelection: (item: Items) => void , numberOfItemsSelected: number}> = ({allItems, handleItemSelection, numberOfItemsSelected}) => {
+
+  const [multiShop, setMultiShop] = useState<Items[]>([])
+
+  const getRandomRarity = () => {
+    const randomNumber: number = Math.floor(Math.random() * 100)
+    if (randomNumber < 70) {
+      return itemRarities.common
+    } else if (randomNumber < 95) {
+      return itemRarities.uncommon
+    } else {
+      return itemRarities.legendary
+    }
+  }
+
+  const getRandomItem = (itemRarityArray: Items[]) => {
+    return Math.floor(Math.random() * itemRarityArray.length)
+  }
```

Comment:
nit: add return types, not going to suggest this in the future, but this is good practice

```suggestion
  const getRandomItem = (itemRarityArray: Items[]): number => {
    return Math.floor(Math.random() * itemRarityArray.length)
  }
```